src/content/en/updates/2018/01/paintapi.md

```diff
@@ -0,0 +1,278 @@
+      type="video/webm; codecs=vp8">
+    <source
+      src="https://storage.googleapis.com/webfundamentals-assets/paintapi/checkercast_x264.mp4"
+      type="video/mp4; codecs=h264">
+  </video>
+</div>
+
+Note: It would be great to parameterize the colors, too, wouldn’t it? The spec
+allows for the `paint()` function to take a list of arguments. This feature is
+not implemented in Chrome yet, as it heavily relies on Houdini’s Properties &
+Values API, which still needs some work before it can ship.
+
+## Browsers that doesn’t support paint worklet
+At the time of writing, only Chrome has paint worklet implemented. While there
+are positive signals from all other browser vendors, there isn’t much progress.
+To keep up to date, check [Is Houdini Ready Yet?](https://ishoudinireadyyet.com)
+on a regular basis. In the meantime, you should make sure to use Progressive
+Enhancement to keep your code running even if there’s no support for paint
+worklet. To make sure things work as expected, you have to adjust your code in
+two places: The CSS and the JS.
+
+Detecting support for paint worklet in JS can be done by checking the `CSS` object:
+
+    if ('paintWorklet' in CSS) {
+      CSS.paintWorklet.addModule('mystuff.js');
+    }
+
+For the CSS side, you have two options. You can use `@supports`:
+
+    @supports (background: paint(id)) {
+      /* ... */
+    }
+
+A more compact trick is to use the fact that CSS invalidates and subsequently
+ignores an entire property declaration if there is an unknown function in it. If
+you specify a property twice — first without paint worklet, and then with the
+paint worklet — you get progressive enhancement:
+
+    textarea {
+      background-image: linear-gradient(0, red, blue);
+      background-image: paint(myGradient, red, blue).
+    }
+
+In browsers _with_ support for paint worklet, the second declaration of
+`background-image` will overwrite the first one. In browsers _without_ support
+for paint worklet, the second declaration is invalid and will be discarded,
+leaving the first declaration in effect.
+
+## Use cases
+There’s many use-cases for Paint Worklets, some of them more obvious than
+others. One of the more obvious ones is using paint worklet to reduce the size
+of your DOM. Oftentimes, elements are added purely to create embellishments
+using CSS. For example, in Material Design Lite the button with the ripple
+effect contains 2 additional `<span>`s to implement the ripple itself. If you
+have a lot of buttons, this can add up to quite a number of DOM elements and can
+lead to degraded performance on mobile. If you [implement the ripple effect
+using paint
+worklet](https://googlechromelabs.github.io/houdini-samples/paint-worklet/ripple/)
+instead, you end up with 0 additional elements and just one paint worklet.
+Additionally, you will end up with something that is much easier to customize
+and parameterize.
+
+Another upside of using paint worklet  is that — in most scenarios — a solution
```

Comment:
can we claim a perf benefit too?

Reply:
I was thinking you're offloading a lot of heavy lifting to a worker. Maybe it's just as fast as a canvas in main thread?

Reply:
For now: yes. Except the the canvas is automatically sized to fit the element’s size and you don‘t have to do any weird serialization dance to get it into CSS.

As I say in the article:

> Chrome is working on moving paint worklets off the main thread so that even long-running paint worklets don’t affect the responsiveness of the main thread.

Reply:
Sweet. I missed that note!